src/hotspot/cpu/aarch64/aarch64_sve.ad

```diff
@@ -3178,25 +3178,192 @@ instruct vtest_anytrue_partial(iRegINoSp dst, vReg src1, vReg src2, pRegGov pTmp
+      __ sve_orr(as_FloatRegister($dst$$reg),
+             as_FloatRegister($src$$reg),
+             as_FloatRegister($src$$reg));
+    }
+    __ sve_cpy(as_FloatRegister($dst$$reg), __ B,
+               as_PRegister($pTmp$$reg), as_Register($val$$reg));
+
+  %}
+  ins_pipe(pipe_slow);
+%}
+
+instruct insertS_fast(vReg dst, vReg src, iRegIorL2I val, immI idx, vReg tmp, pRegGov pTmp, rFlagsReg cr)
+%{
+  predicate(UseSVE > 0 && n->as_Vector()->length() <= 32 &&
+            n->bottom_type()->is_vect()->element_basic_type() == T_SHORT);
+  match(Set dst (VectorInsert (Binary src val) idx));
+  effect(TEMP tmp, TEMP pTmp, KILL cr);
+  ins_cost(4 * SVE_COST);
+  format %{ "sve_index $tmp, H, -16, 1\n\t"
+            "sve_cmpeq $pTmp, $tmp, ($idx-#16) // shift from [0, 31] to [-16, 15]\n\t"
+            "sve_orr $dst, $src, $src\n\t"
+            "sve_cpy $dst, $pTmp, $val\n\t# insert into vector (S)" %}
+  ins_encode %{
+    __ sve_index(as_FloatRegister($tmp$$reg), __ H, -16, 1);
+    __ sve_cmpeq(as_PRegister($pTmp$$reg), __ H, ptrue,
+                 as_FloatRegister($tmp$$reg), (int)($idx$$constant) - 16);
+    // If src and dst are the same reg, this move is not needed.
+    if (as_FloatRegister($dst$$reg) != as_FloatRegister($src$$reg)) {
+      __ sve_orr(as_FloatRegister($dst$$reg),
+             as_FloatRegister($src$$reg),
+             as_FloatRegister($src$$reg));
+    }
+    __ sve_cpy(as_FloatRegister($dst$$reg), __ H,
+               as_PRegister($pTmp$$reg), as_Register($val$$reg));
+
+  %}
+  ins_pipe(pipe_slow);
+%}
+
+instruct insertI_fast(vReg dst, vReg src, iRegIorL2I val, immI idx, vReg tmp, pRegGov pTmp, rFlagsReg cr)
+%{
+  predicate(UseSVE > 0 && n->as_Vector()->length() <= 32 &&
+            n->bottom_type()->is_vect()->element_basic_type() == T_INT);
+  match(Set dst (VectorInsert (Binary src val) idx));
+  effect(TEMP tmp, TEMP pTmp, KILL cr);
+  ins_cost(4 * SVE_COST);
+  format %{ "sve_index $tmp, S, -16, 1\n\t"
+            "sve_cmpeq $pTmp, $tmp, ($idx-#16) // shift from [0, 31] to [-16, 15]\n\t"
+            "sve_orr $dst, $src, $src\n\t"
+            "sve_cpy $dst, $pTmp, $val\n\t# insert into vector (I)" %}
+  ins_encode %{
+    __ sve_index(as_FloatRegister($tmp$$reg), __ S, -16, 1);
+    __ sve_cmpeq(as_PRegister($pTmp$$reg), __ S, ptrue,
+                 as_FloatRegister($tmp$$reg), (int)($idx$$constant) - 16);
+    // If src and dst are the same reg, this move is not needed.
+    if (as_FloatRegister($dst$$reg) != as_FloatRegister($src$$reg)) {
+      __ sve_orr(as_FloatRegister($dst$$reg),
+             as_FloatRegister($src$$reg),
+             as_FloatRegister($src$$reg));
+    }
+    __ sve_cpy(as_FloatRegister($dst$$reg), __ S,
+               as_PRegister($pTmp$$reg), as_Register($val$$reg));
+
+  %}
+  ins_pipe(pipe_slow);
+%}
+
+instruct insertL_fast(vReg dst, vReg src, iRegL val, immI idx, vReg tmp, pRegGov pTmp, rFlagsReg cr)
+%{
+  predicate(UseSVE > 0 && n->as_Vector()->length() <= 32 &&
```

Comment:
Can `n->as_Vector()->length() <= 32` be removed? Or maybe an assertion is better.

Reply:
Yes . The reason is the same as last comment. https://github.com/openjdk/panama-vector/pull/56#discussion_r607437523
 Or we COULD split insertL and insertD from other ones ?

Reply:
This seems reasonable to me. Agree to split L/D from others.